src/nomad_measurements/utils.py

```diff
@@ -371,28 +373,32 @@ def write_file(self):
         """
         if self.nexus:
             try:
+                delete_entry_file(archive=self.archive, mainfile=self.data_file)
```

Comment:
How can you ensure the caller is not appending new data to existing data?
How can you ensure the caller always provides a complete file?
When the existing file is deleted, how can you tell what contents are valid and may need to be retained?